apps/algorithm/search/src/server.py

```diff
@@ -1,8 +1,117 @@
-from fastapi import FastAPI, Response
+from enum import Enum
+from typing import List
+
+from fastapi import FastAPI
+from pydantic import BaseModel, Field
 
 app = FastAPI()
 
 
-@app.get("/api/search")
-async def home(response: Response):
-    return "Hello World"
+# define enums
+class Status(str, Enum):
+    AVAILABLE = "AVAILABLE"
+    SOLD = "SOLD"
+
+
+class SearchType(str, Enum):
+    LISTINGS = "LISTINGS"
+    SELLERS = "SELLERS"
+
+
+class Sort(str, Enum):
+    RELEVANCE = "RELEVANCE"
+    PRICE_ASC = "PRICE_ASC"
+    PRICE_DESC = "PRICE_DESC"
+    LISTED_TIME_ASC = "LISTED_TIME_ASC"
+    LISTED_TIME_DESC = "LISTED_TIME_DESC"
+    DISTANCE_ASC = "DISTANCE_ASC"
+    DISTANCE_DESC = "DISTANCE_DESC"
+
+
+# Define Pydantic models
+class ListingSummary(BaseModel):
+    listingID: str = Field(..., example="A23F29039B23")
+    sellerID: str = Field(..., example="A23F29039B23")
+    sellerName: str = Field(..., example="John Doe")
+    title: str = Field(..., example="Used Calculus Textbook")
+    description: str = Field(..., example="No wear and tear, drop-off available.")
+    price: float = Field(..., example=50)
+    dateCreated: str = Field(..., example="2024-05-23T15:30:00Z")
+    imageUrl: str = Field(..., example="image URL for first Image")
+
+
+class Listing(BaseModel):
+    listingId: str = Field(..., example="abc123")
+    sellerId: str = Field(..., example="seller456")
+    sellerName: str = Field(..., example="billybobjoe")
+    title: str = Field(..., example="High-Performance Laptop")
+    description: str = Field(
+        ..., example="A powerful laptop suitable for gaming " "and professional use."
+    )
+    price: float = Field(..., example=450.00)
+    location: dict = Field(..., example={"latitude": 45.4215, "longitude": -75.6972})
+    status: str = Field(..., example="AVAILABLE")
+    dateCreated: str = Field(..., example="2024-05-22T10:30:00Z")
+    imageUrl: str = Field(..., example="https://example.com/image1.jpg")
+
+
+@app.get("/api/search", response_model=List[ListingSummary])
+async def search(
+    authorization: str,
+    query: str,
+    latitude: float,
+    longitude: float,
+    page: int,
+    limit: int,
```

Comment:
These should definitely have defaults. My assumption would be:

```suggestion
    page: int = 1,
    limit: int = 20,
```